mesonbuild/backend/backends.py

```diff
@@ -291,7 +327,7 @@ def get_build_dir_include_args(self, target, compiler, *, absolute_path=False):
                 curdir = '.'
         return compiler.get_include_args(curdir, False)
 
-    def get_target_filename_for_linking(self, target):
+    def get_target_filename_for_linking(self, target: T.Union[build.Target, build.CustomTargetIndex]) -> T.Optional[str]:
```

Comment:
Is it really out of the question that not a `build.CustomTarget` can be passed here? There is an `isinstance` check for it below.

Same with the other functions below.

Reply:
Custom target is a Target (though not a BuildTarget), but CustomTargetIndex is its own thing.

Reply:
Oh, right. Makes sense.